src/core/manager.c

```diff
@@ -736,6 +736,10 @@ int manager_new(UnitFileScope scope, unsigned test_run_flags, Manager **_m) {
         if (r < 0)
                 return r;
 
+        r = set_ensure_allocated(&m->pending_finished_jobs, NULL);
```

Comment:
Sounds like this should be allocated only when used, i.e. right before the `set_put` above.